xcodeproj/internal/xcodeproj_macro.bzl

```diff
@@ -242,6 +250,15 @@ def xcodeproj(
             hash of all transitive dependencies of the targets specified in the
             `top_level_targets` argument with the `"simulator"`
             `target_environment`, even if they aren't watchOS targets.
+        xcode_configurations: Optional. A `dict` mapping Xcode configuration
+            names to transition settings dictionaries. For example,
+            `{"Dev": {"//command_line_option:compilation_mode": "dbg"}, "AppStore": {"//command_line_option:compilation_mode": "opt"}}`,
```

Comment:
This example is useful, but more examples for common cases would also be good. You could say that's out of scope here and you do link out to the bazel docs which is great, but maybe being able to answer "how can I specify a bazel config (aka a `--config $foo`) in an Xcode configuration" would be really useful.

Maybe I'm misunderstanding something and a transition can't apply an arbitrary bazel config in that way.

Reply:
Correct, you can't use `--config`. The main reason I linked out, is because the rules around what is supported in a transition is out of our hands, and it probably will change with time.